GSA-TTS/FAC
# Maintenance mode toggle

There is no direct issue linked to this PR - but it addresses recent dev conversation about implementing a toggle that can be switched via management command to disable/re-enable the FAC application in order to carry out rare but necessary processes such as historical data migrations.

## How to test

Make sure you run a clean version of the container and ensure that it can still run some e2e tests successfully.
Start up a shell, then run through the following tests:
1. `python manage.py maintenance_mode` (expectation is no response - the command needs a parameter)
2. `python manage.py maintenance_mode --on` (expectation is that maintenance mode will activate, and a flag will upload to `/runtime` in S3)
3. Validate the application will redirect you to the `/maintenance` slug on every request.
4. `python manage.py maintenance_mode --off` (expectation is that maintenance mode will deactivate, and a flag will be removed from `/runtime` in S3 if it exists)
5. Validate the application will start accepting requests without redirecting you to the `/maintenance` slug.

## PR Checklist: Submitter


## PR Checklist: Reviewer


The larger the PR, the stricter we should be about these points.

## Pre Merge Checklist: Merger
